src/Microsoft.VisualStudio.Threading.Tests/TestBase.cs

```diff
@@ -324,5 +325,37 @@ protected void ExecuteOnDispatcher(Func<Task> action)
             }
 #endif
         }
+
+        /// <summary>
+        /// Executes the specified test method in its own process, offering maximum isolation from ambient noise from other threads
+        /// and GC.
+        /// </summary>
+        /// <param name="testMethodName">The name of the test method.</param>
+        /// <returns>
+        /// A task whose result is <c>true</c> if test execution is already isolated and should therefore proceed with the body of the test,
+        /// or <c>false</c> after the isolated instance of the test has completed execution.
+        /// </returns>
+        /// <exception cref="Xunit.Sdk.XunitException">Thrown if the isolated test result is a Failure.</exception>
+        /// <exception cref="SkipException">Thrown if on a platform that we do not yet support test isolation on.</exception>
+        protected Task<bool> ExecuteInIsolationAsync([CallerMemberName] string testMethodName = null)
```

Comment:
Would it make sense to have an xUnit extension, like [Isolated] that would make xUnit run the attributed test method on the isolated host?